kong/db/strategies/postgres/connector.lua

```diff
@@ -552,6 +553,11 @@ function _mt:query(sql, operation)
 
   res, err, partial, num_queries = conn:query(sql)
 
+  if (kong and kong.configuration and kong.configuration.log_level == "debug")
+          and in_admin_api and operation == "write"
+          and res and res[1] and res[1]._pg_transaction_id then
+    kong.response.set_header('X-Kong-Test-Transaction-ID', res[1]._pg_transaction_id)
```

Comment:
I feel a bit uncomfortable to set response header in db connector. Should we just set `ngx.ctx.kong_transaction_id` or something here, and the header for example in here: https://github.com/Kong/kong/blob/master/kong/init.lua#L1838

Reply:
I've followed your suggestion and added a call to `kong_global.get_current_transaction_id()` to `serve_content` in kong/init.lua.  This is now gone.